ООП Лабораторная работа 2

SurkovaSS opened the pull request.

SurkovaSS committed:
ООП Лабораторная работа 2

zagranet reviewed (changes requested):
- Нет аннотации классов с описанием параметров.
- Нет аннотации методов.
- В конструкторе, в качестве параметра по умолчанию Вы использовали пустой список. Нужно исправить на значение по умолчанию None.
- В методе get_next_book_id() нужно использовать проверку if not self.books:
- В методе get_index_by_book_id() параметр value - это книга.

SurkovaSS committed:
Update Класс_Book.py

Добавлены аннотации классов с описанием параметров, а также аннотации методов.

SurkovaSS committed:
Update Класс_Library.py

1) Добавлены аннотации классов с описанием параметров, а также аннотации методов.
В предыдущей лабораторной работе было явно указано, что везде необходимы аннотации с примерами для doctest`а, а в этой об этом не было ни слова. Не могли бы вы уточнить, впредь во всех лабораторных необходимо будет писать аннотации?
2) В конструкторе, в качестве параметра по умолчанию изменено значение с пустого списка на None.
Немного непонятно почему должен быть именно None, хотя в задании указано "Если пользователь его не передал, то библиотека инициализируется с пустым списком книг.". Но тем не менее параметр был изменен.
3) В методе get_next_book_id() изменена проверка на "if not self.books:"
4) В методе get_index_by_book_id() параметр value - это книга... но это никак не противоречит тому факту, что должен выдоваться номер книги с из списка библиотечных книг с идентификатором, значение которого равно value.
